.github/workflows/milestone.yml

```diff
@@ -17,10 +16,21 @@ jobs:
           ./mvnw -v
           echo "MILESTONE_NUMBER=$(./mvnw -q -Dexec.executable=echo -Dexec.args='${project.version}' -N exec:exec | cut -d- -f1)" >> $GITHUB_ENV
       - name: Set milestone to PR
-        uses: actions/github-script@v3
+        uses: actions/github-script@v4
         with:
           github-token: ${{secrets.GITHUB_TOKEN}}
           script: |
+            // Get pull request number from commit sha
+            const pr_response = await github.repos.listPullRequestsAssociatedWithCommit({ owner: context.repo.owner, repo:context.repo.repo, commit_sha: context.sha })
```

Comment:
Does this list only merged PRs?
There can be multiple PRs with same commit (for example when someone splits a PR into two or creates a PR directly from Trino master to run tests against Glue/MemSQL).

Reply:
From docs (https://docs.github.com/en/rest/reference/repos#list-pull-requests-associated-with-a-commit)

> Lists the merged pull request that introduced the commit to the repository. If the commit is not present in the default branch, additionally returns open pull requests associated with the commit. The results may include open and closed pull requests.

Reply:
Oh, looks like it works because it lists merged. Only if no merged PR is found it will list open/closed ones.
So since the commit we are searching is already in master it'll always be found and the fallback won't trigger.
Weird API design.

IMO it deserves some comment.

Reply:
@hashhar Added comments.